library/src/main/java/com/getstream/sdk/chat/view/ChannelListView.java

```diff
@@ -126,8 +127,16 @@ public void setOnChannelClickListener(ChannelClickListener l) {
 
     public void setOnLongClickListener(ChannelClickListener l) {
         this.channelLongClickListener = l;
-        if (adapter != null) {
+        if (adapter == null) return;
+
+        if (l != null) {
             adapter.setChannelLongClickListener(l);
+        }else{
+            adapter.setChannelLongClickListener(channel ->
```

Comment:
let's not bundle this in the SDK; if `channelLongClickListener` is null we should do nothing. I suggest moving the whole `ChannelMoreActionDialog` code to the sample app